Services/Mentor/OJS.Services.Mentor.Business/Implementations/MentorBusinessService.cs

```diff
@@ -0,0 +1,464 @@
+    {
+        this.userMentorData = userMentorData;
+        this.mentorPromptTemplateData = mentorPromptTemplateData;
+        this.openAiClient = new OpenAIClient(configuration.GetSectionWithValidation<MentorConfig>().ApiKey);
+        this.httpClientFactory = httpClientFactory;
+        this.settingData = settingData;
+    }
+
+    public async Task<ConversationResponseModel> StartConversation(ConversationRequestModel model)
+    {
+        var settings = await this.settingData
+            .GetQuery(s => s.Name.StartsWith(Mentor))
+            .ToDictionaryAsync(k => k.Name, v => v.Value);
+
+        var userMentor = await this.userMentorData
+            .GetQuery(um => um.Id == model.UserId)
+            .FirstOrDefaultAsync();
+
+        if (userMentor is null)
+        {
+            userMentor = new UserMentor
+            {
+                Id = model.UserId,
+                CreatedOn = DateTime.UtcNow,
+                ModifiedOn = DateTime.UtcNow,
+                RequestsMade = 0,
+                QuotaLimit = null,
+                QuotaResetTime = DateTime.UtcNow.AddMinutes(GetNumericValue(settings, nameof(MentorQuotaResetTimeInMinutes))),
+            };
+
+            await this.userMentorData.Add(userMentor);
+            await this.userMentorData.SaveChanges();
+        }
+
+        if (DateTime.UtcNow >= userMentor.QuotaResetTime)
+        {
+            userMentor.RequestsMade = 0;
+            userMentor.QuotaResetTime = DateTime.UtcNow.AddMinutes(GetNumericValue(settings, nameof(MentorQuotaResetTimeInMinutes)));
+        }
+
+        if (userMentor.RequestsMade > (userMentor.QuotaLimit ?? GetNumericValue(settings, nameof(MentorQuotaLimit))))
+        {
+            throw new BusinessServiceException($"You have reached your message limit, please try again after {GetTimeUntilNextMessage(userMentor.QuotaResetTime)}.");
+        }
+
+        if (model.ConversationMessages.All(cm => cm.Role != ChatMessageRole.System))
+        {
+            var template = await this.mentorPromptTemplateData
+                .GetQuery()
+                .FirstOrDefaultAsync();
+
+            var wordFiles = model.ProblemResources
+                .Where(pr =>
+                       pr is { File: not null, FileExtension: Docx } ||
+                       pr.Link is not null && pr.Link.Split('.').Last().Equals(Docx, StringComparison.Ordinal))
+                .ToList();
+
+            var problemResource = wordFiles
+                                    .FirstOrDefault(pr => pr.ProblemId == model.ProblemId) ??
+                                wordFiles.FirstOrDefault();
+
+            if (problemResource is null)
+            {
+                throw new BusinessServiceException(DocumentNotFoundOrEmpty);
+            }
+
+            var file = problemResource.File ??
+                (problemResource.Link is not null ?
+                await this.DownloadDocument(problemResource.Link) :
+                []);
```

Comment:
Formatting: `?` is on new line followed by the logic, same applies for `:`

Reply:
Done.